appcatalog/frontendUpload.go

```
@@ -63,17 +64,23 @@ func UploadToFrontend(uploadURI string, zapFile string, appName string, sessionI
 
 	log.Printf("Express frontend returned status code %v.", response.StatusCode)
 
-	if response.StatusCode == http.StatusOK {
-		log.Println("The app has been uploaded to the frontend successfully.")
-	} else {
-		return "", fmt.Errorf("Uploading failed, the frontend returned status code %v", response.StatusCode)
-	}
-
 	// The frontend returns a link which can be used to poll the upload status.
 	// {
 	//   "links": {
 	//     "progress": "https://fireball-dev.travix.com/upload/progress?sessionId=123`"
 	//   }
 	// }
-	return responseObject["links"]["progress"], nil
+
+	progressUri := responseObject["links"]["progress"]
+	if len(strings.TrimSpace(progressUri)) == 0 {
+		return "", fmt.Errorf("Uploading failed, the app catalog did not return a valid response")
+	}
+
+	if response.StatusCode == http.StatusOK {
```

@markvincze I have fixed it to first check the status code and fail early if different than 200.  I've also added an extra test case.